src/include/function/cast/cast_functions.h

```diff
@@ -87,27 +88,85 @@ inline std::string CastToString::castToStringWithVector(
 
 template<typename SRC, typename DST>
 static inline void numericDownCast(SRC& input, DST& result, const std::string& dstTypeStr) {
-    if (input < std::numeric_limits<DST>::min() || input > std::numeric_limits<DST>::max()) {
+    if ((signed)input < std::numeric_limits<DST>::min() ||
+        (signed)input > std::numeric_limits<DST>::max()) {
+        auto a = std::numeric_limits<DST>::min();
+        auto b = std::numeric_limits<DST>::max();
         throw common::RuntimeException(
             "Cast failed. " + std::to_string(input) + " is not in " + dstTypeStr + " range.");
     }
     result = (DST)input;
 }
 
+template<typename T>
+static inline void stringCastNum(
+    char*& input, const size_t& len, T& result, const std::string& type) {
```

Comment:
Why use reference to a pointer? We don't use reference for primitive types when passing parameters.